### DISCUSSION TITLE
How to style child components

### DISCUSSION THREAD

**Discussion by LuisPacx** [chakra-ui/chakra-ui#4159] [Q&A] [answered]
Hello there!

I was trying to implement the following code from "styled-components" using chakra, since yesterday I couldn't figure it out:

```
export const CardImage = styled.div`
    position: relative;
`

export const Card = styled.div`
    position: relative;
    ${CardImage}{
        max-height: 60%;
        overflow: hidden;
    }
`
```

This "hack" help me to speed up many designs using styled-components and since I change to Chakra-ui I can't make this work that way, and now I'm asking myself if there's some way to make this work using Chakra? 

I have some ideas to solve the problem but it's more "css" way and I think it should be something similar in chakra-ui. I will keep diving into the documentation.

Thank you so much for reading and looking forward to your workaround!

High vibes,
Luis Kunz

**UPDATE:**

I figure it out that I could use the emotion babel macros on this way:

```
const VerticalCardOverlay = chakra(`div`, {
  baseStyle: {
    position: `absolute`,
    top: 0,
    right: `-1px`,
    bottom: `-1px`,
    left: `-1px`,
    zIndex: 2,
    bgGradient: `linear(to-tr, primary.600, primary.800)`,
    opacity: `0.5`,
  },
})

const VerticalCard = chakra(`div`, {
  baseStyle: {
    overflow: `hidden`,
    display: `flex`,
    position: `relative`,
    justifyContent: `center`,
    alignItems: `center`,
    width: `100%`,
    height: [`20rem`, `25rem`, null, `40rem`],
    boxShadow: `lg`,
    [`:hover ${VerticalCardOverlay}`]: {
      opacity: `0`,
    },
  },
})
```

**Comment by LuisPacx** [comment]
SOLUTION:

Even though I couldn't make it work exactly as I would in styled-components, I could use VerticalCardOverlay inside of VerticalCard as a child component and apply:

`<VerticalCardOverlay _hover={{ opacity: 1 }} />`

It works pretty fine and isn't a bad way to solve it. I hope it can help somebody with the same requirements.

Any other solution which could improve speed/simplicity I would be pleased to hear!

Best regards,
Luis

**Comment by anubra266** [comment, marked as answer]
You couled export  `styled` from emotion, and do it the same way.
```js
import styled from '@emotion/styled'
```

**Reply by LuisPacx** [reply]
Wow! I didn't know that I could import `styled` from emotion, that's a game-changer! Thanks, @anubra266 amazing!

**Reply by anubra266** [reply]
Happy to help...Please mark the right answer, to help anyone that later comes across this.

**Reply by LuisPacx** [reply]
Hey @anubra266 I tried using styled but it is not possible due the following reason:
`Component selectors can only be used in conjunction with @emotion/babel-plugin.`

Seems like isn't possible to add a component selector inside styled components, just in case I will leave you here the code I tried:

```
const MotionTitle = styled.p`
  color: white;
`

const Container = styled.div`
  weight: 100%;
  height: 20rem;
  position: relative;
  box-shadow: lg;
  padding: 1rem;
  align-items: flex-end;
  justify-content: center;
  overflow: hidden;
  &:hover {
    ${MotionTitle} {
      opacity: 0.5;
    }
  }
`
```

Any ideas?

**Reply by anubra266** [reply]
Oh...you need the babel plugin.

So let's try the chakraesque way

**Reply by anubra266** [reply]
I haven't used component selectors before, but I've played around and love the concept. I'll open an issue, to get the workings.

**Reply by LuisPacx** [reply]
> I haven't used component selectors before, but I've played around and love the concept. I'll open an issue, to get the workings.

That's an amazing idea, happy to hear about it!

Just for curiosity, what do you use instead of component selectors? Raw CSS or any Chakraesque way maybe?

**Reply by anubra266** [reply]
Well It depends on what you're trying to achieve. Could you share, so I can streamline my help.

**Reply by LuisPacx** [reply]
For sure!

Description: Card component with headline at bottom
onHover: Move Headline centered with the new description below. Additionally, I have a simple Overlay background.

```
const Card: React.FC<ICardProps> = ({ children, image, height, overlay, textOverlay, ...otherProps }) => (
  <Flex
    w="100%"
    h={height || [`10rem`, null, `15rem`, `20rem`]}
    pos="relative"
    boxShadow="lg"
    p="1rem"
    alignItems="flex-end"
    justifyContent="center"
    overflow="hidden"
    {...otherProps}
  >
    <Image src={image} pos="absolute" objectFit="cover" h="100%" w="100%" top="0" bottom="0" left="0" right="0" />
    {children}
    {overlay && <Overlay />}
    {textOverlay && <TextOverlay />}
  </Flex>
)
```

The component is in the "static" phase, trying to figure out how to accomplish using chakra-ui. I'm having plenty of fun with it, additionally, the Chakra's documentation is quite complete.

**Reply by anubra266** [reply]
One important thing. If you want a child to know when it's parent is hovered. you use the `role="group"` and `_groupHover` props
```js
<Parent role="group">
      <Child _groupHover={{
                color:'red'
           }}>
           Child
      </Child>
</Parent>
```
Here Child becomes red, when Parent is hovered.
In your case, `Flex` should take the `role`, then any of the children can be customized with the `groupHover`



Example Here
https://chakra-ui.com/docs/features/style-props#pseudo

**Reply by LuisPacx** [reply]
You should saw my face when I read that! That would make it, it's just amazing how Chakra handles so much stuff in such a clean way! Thank you so much for your help! 

This issue can be happily closed.

**Reply by anubra266** [reply]
Cheers and good luck on your project.

**Comment by chief-austinc** [comment]
curious if there are any updates to the component selector pattern from styled/emotion

- https://styled-components.com/docs/advanced#referring-to-other-components
- https://emotion.sh/docs/styled#targeting-another-emotion-component
